pkg/utils/has/controller.go

```diff
@@ -243,6 +244,11 @@ func (h *SuiteController) CreateComponent(applicationName, componentName, namesp
 		component = h.refreshComponentForErrorDebug(component)
 		return nil, fmt.Errorf("timed out when waiting for component %s to be ready in %s namespace. component: %s", componentName, namespace, utils.ToPrettyJSONString(component))
 	}
+	if containerImage == "" {
+		if err = utils.WaitUntil(h.CheckForImageAnnotation(component), time.Minute*1); err != nil {
```

Comment:
this is 4 times in the code, would be nicer to have it as function.

Reply:
I see 4 places where we are creating component with different variations and this part of the code is checking if image annotations exists after the component creation.We need to check this in all the 4 places where we are creating component. Do you mean we can further improve this to create a another function? Could you please explain a bit more.